scripts/tensorize.sh

```diff
@@ -119,20 +119,30 @@ START_TIME=$(date +%s)
 MIN_SAMPLE_ID=$SAMPLE_IDS_START
 MAX_SAMPLE_ID=$SAMPLE_IDS_END
 
-# We want to get the zip folder that was passes to recipes.py - look for the --zip_folder argument and extract the value passed after that
-ZIP_FOLDER=$(echo ${PYTHON_ARGS} | sed 's/--zip_folder \([^ ]*\).*/\1/')
-if [ ! -e $ZIP_FOLDER ]; then
-    echo "ERROR: Zip folder passed was not valid, found $ZIP_FOLDER but expected folder path." 1>&2
-    exit 1
-fi
-
 # create a directory in the /tmp/ folder to store some utilities for use later
 mkdir -p /tmp/ml4h
-# Write out a file with the ids of every sample in the input folder
-echo "Gathering list of input zips to process between $MIN_SAMPLE_ID and $MAX_SAMPLE_ID, this takes several seconds..."
-find $ZIP_FOLDER -name '*.zip' | xargs -I {} basename {} | cut -d '_' -f 1 \
-                               | awk -v min="$MIN_SAMPLE_ID" -v max="$MAX_SAMPLE_ID" '$1 > min && $1 < max' \
-                               | sort | uniq > /tmp/ml4h/sample_ids_trimmed.txt
+
+
+# We want to get the zip folder that was passes to recipes.py - look for the --zip_folder argument and extract the value passed after that
+# Or, if tensorizing pngs, we can look for sample_ids as the first column of the manifest file
+ZIP_FOLDER=$(echo ${PYTHON_ARGS} | sed -n 's/.*--zip_folder \([^ ]*\).*/\1/p')
+MANIFEST=$(echo ${PYTHON_ARGS} | sed -n 's/.*--app_csv \([^ ]*\).*/\1/p')
+
+# Write out a file with the ids of every sample in the input folder (or csv manifest if tensorizing pngs)
+if [ -e "$ZIP_FOLDER" ]; then
+  echo "Gathering list of input zips to process between $MIN_SAMPLE_ID and $MAX_SAMPLE_ID, this takes several seconds..."
+  find $ZIP_FOLDER -name '*.zip' | xargs -I {} basename {} | cut -d '_' -f 1 \
+                                 | awk -v min="$MIN_SAMPLE_ID" -v max="$MAX_SAMPLE_ID" '$1 > min && $1 < max' \
+                                 | sort | uniq > /tmp/ml4h/sample_ids_trimmed.txt
+elif [ -e "$MANIFEST" ]; then
```

Comment:
Thanks! You convinced me to fix this up a bit better =) I've made some changes that more explicity refer to whether a 'tensorize' or 'tensorize_pngs' job is running, so that the right data is pulled from. There are also modes to 'tensorize_ecg_pngs' and 'tensorize_partners', but Sam says no one has done that for a while and we can just have the script fail in those cases.